### ISSUE TITLE
[infra-proxy-service] Could not get clients: grpc: received message larger than max

### ISSUE THREAD

**Issue by vsingh-msys** [chef/automate#4210]
## Describe the bug
With large data set we have come across with the grpc limits error and getting `Could not get clients: grpc: received message larger than max (39700052 vs. 4194304)`

Seems it pulling all data in one go need to add pagination in all APIs.

## To Reproduce
- go to `a2-cs-single-local-inplace-upgrade-acceptance.cd.chef.co`
- Dashboard >> Infrastructure >> Chef Servers >> Clients

<img width="637" alt="fun" src="https://user-images.githubusercontent.com/44496937/89816738-b77f3080-db64-11ea-9d5a-409c19ca50c5.png">

**Comment by btm** [comment]
@vsingh-msys Are you getting 39mb of data from `GET /organizations/NAME/clients` or is it all the data coming back from many calls to `GET /organizations/NAME/clients/NAME` ?

**Comment by vsingh-msys** [comment]
@btm not sure yet, seem environment contains 64k nodes so might it has the same number of clients. trying to replicate the issue on local dev.

**Comment by vsingh-msys** [comment]
@btm at local dev environment, have put 110k records of clients but still working as expected. will keep an eye on the given environment.

**Comment by vsingh-msys** [comment]
Setup the knife config with provided acceptance environment. It has 132K clients records and the response hash size almost **64MB** of data `ObjectSpace.memsize_of(return_value)`. Also wrote a response into file and size of file about **206MB**.

```
bundle exec knife client list -VVV
INFO: Using configuration from /chef-workstation/.chef/config.rb
TRACE: Chef::HTTP calling Chef::HTTP::JSONInput#handle_request
TRACE: Chef::HTTP calling Chef::HTTP::JSONOutput#handle_request
TRACE: Chef::HTTP calling Chef::HTTP::CookieManager#handle_request
TRACE: Chef::HTTP calling Chef::HTTP::Decompressor#handle_request
TRACE: Chef::HTTP calling Chef::HTTP::Authenticator#handle_request
TRACE: Signing the request as test-admin
TRACE: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_request
TRACE: Chef::HTTP calling Chef::HTTP::APIVersions#handle_request
TRACE: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_request
TRACE: Initiating GET to https://a2-cs-single-local-inplace-upgrade-acceptance.cd.chef.co/organizations/test/clients
TRACE: ---- HTTP Request Header Data: ----
TRACE: Accept: application/json
TRACE: Accept-Encoding: gzip;q=1.0,deflate;q=0.6,identity;q=0.3
TRACE: X-Ops-Server-API-Version: 2
TRACE: X-OPS-SIGN: algorithm=sha1;version=1.1;
TRACE: X-OPS-USERID: test-admin
TRACE: X-OPS-TIMESTAMP: 2020-08-18T21:03:36Z
TRACE: X-OPS-CONTENT-HASH: 2jmj7l5rSw0yVb/vlWAYkK/YBwk=
TRACE: X-OPS-AUTHORIZATION-1: AofG4GH9YuTcDQK/bXMoWXW5msYPUsW0BIwdSfANi7gx2csX6nk9J+mc2BWm
TRACE: X-OPS-AUTHORIZATION-2: 75r+cl5Bw3Z+2VP0Gt1ubw6omgVlCX2vpxNrJ/BLdtyXv9VLXPwOIwxO7Wky
TRACE: X-OPS-AUTHORIZATION-3: mjNzAQqjiLmu/K4Kv/imj5o1S4Ks0dwulWbZr7hqcKWWy/vCU8fi76iF2C7K
TRACE: X-OPS-AUTHORIZATION-4: 64hSWsi7HgezVVCJS0p9SSd8GxllwKZPWMCk8rHy5AyoLZzmuGjKO0FM65R6
TRACE: HOST: a2-cs-single-local-inplace-upgrade-acceptance.cd.chef.co:443
TRACE: X-REMOTE-REQUEST-ID: 895c78be-d738-4b5e-926c-c832e30d7460
TRACE: ---- End HTTP Request Header Data ----
TRACE: ---- HTTP Status and Header Data: ----
TRACE: HTTP 1.1 200 OK
TRACE: server: nginx
TRACE: date: Tue, 18 Aug 2020 21:04:21 GMT
TRACE: content-type: application/json
TRACE: transfer-encoding: chunked
TRACE: connection: close
TRACE: x-ops-server-api-version: {"min_version":"0","max_version":"2","request_version":"2","response_version":"2"}
TRACE: x-ops-api-info: flavor=cs;version=12.0.0;oc_erchef=14.0.22
TRACE: content-encoding: gzip
TRACE: cache-control: no-cache, no-store, must-revalidate
TRACE: x-xss-protection: 1; mode=block
TRACE: strict-transport-security: max-age=63072000; includeSubDomains
TRACE: x-content-type-options: nosniff
TRACE: ---- End HTTP Status/Header Data ----
TRACE: Chef::HTTP calling Chef::HTTP::ValidateContentLength#handle_response
TRACE: HTTP server did not include a Content-Length header in response, cannot identify truncated downloads.
TRACE: Chef::HTTP calling Chef::HTTP::APIVersions#handle_response
TRACE: Chef::HTTP calling Chef::HTTP::RemoteRequestID#handle_response
TRACE: Chef::HTTP calling Chef::HTTP::Authenticator#handle_response
TRACE: Chef::HTTP calling Chef::HTTP::Decompressor#handle_response
TRACE: Decompressing gzip response
TRACE: Chef::HTTP calling Chef::HTTP::CookieManager#handle_response
TRACE: Chef::HTTP calling Chef::HTTP::JSONOutput#handle_response
TRACE: Chef::HTTP calling Chef::HTTP::JSONInput#handle_response

```

**Comment by btm** [comment]
https://a2-cs-single-local-inplace-upgrade-acceptance.cd.chef.co/infrastructure/chef-servers/auto-deployed-chef-server/organizations/auto-deployed-chef-org mostly just spins for me, but eventually puts up some errors.

@vsingh-msys 1,319,929 is a lot of clients for one organization. If I had to venture a wild guess I'd say we wouldn't see more than 100k clients/nodes per organization.

I did somehow get this error too, no big deal, but it's a little messy:
![Screenshot from 2020-08-18 12-47-01](https://user-images.githubusercontent.com/4381/90576493-65559500-e18c-11ea-8685-3825c4df5ba9.png)

I'm running this loop against the chef server right now as I wanted to get all the the clients on disk. It's taking 1-2 seconds per client. 
```
for c in $(knife client list); do
  echo $c
  knife client show $c -F json > $c.json
done
```
And java/elasticsearch is pegged to 100% of a CPU on the system, which is surely the timeout. 

This client list / grpc issue originally reported here is possibly entirely because this system is being pegged constantly by the a2-cs-chef-load-acceptance.cd.chef.co instance which is running https://github.com/chef/chef-load against it.

/cc @thomascate @andy-dufour
